src/hooks/loadables/useLoadTxHistory.ts

```diff
@@ -1,20 +1,39 @@
 import { useEffect } from 'react'
-import { getTransactionHistory, type TransactionListPage } from '@gnosis.pm/safe-react-gateway-sdk'
+import {
+  getTransactionHistory,
+  type SafeIncomingTransfersResponse,
+  type SafeModuleTransactionsResponse,
+  type SafeMultisigTransactionsResponse,
+  type TransactionListPage,
+} from '@gnosis.pm/safe-react-gateway-sdk'
+import { useRouter } from 'next/router'
 import useAsync, { type AsyncResult } from '../useAsync'
 import { Errors, logError } from '@/services/exceptions'
 import useSafeInfo from '../useSafeInfo'
+import { getFilteredTxHistory, hasTxFilterQuery } from '@/components/transactions/TxFilterForm/utils'
 
 export const useLoadTxHistory = (): AsyncResult<TransactionListPage> => {
+  const router = useRouter()
+
   const { safe, safeAddress, safeLoaded } = useSafeInfo()
   const { chainId, txHistoryTag } = safe
 
   // Re-fetch when chainId/address, or txHistoryTag change
-  const [data, error, loading] = useAsync<TransactionListPage | undefined>(
+  const [data, error, loading] = useAsync<
+    | TransactionListPage
+    | SafeIncomingTransfersResponse
+    | SafeMultisigTransactionsResponse
+    | SafeModuleTransactionsResponse
+    | undefined
+  >(
     async () => {
       if (!safeLoaded) return
-      return getTransactionHistory(chainId, safeAddress)
+
+      return hasTxFilterQuery(router.query)
+        ? getFilteredTxHistory(chainId, safeAddress, router.query)
+        : getTransactionHistory(chainId, safeAddress)
     },
-    [safeLoaded, chainId, safeAddress, txHistoryTag],
+    [safeLoaded, chainId, safeAddress, txHistoryTag, router.query],
     false,
   )
 
```

Comment:
You should not modify this hook. It's for polling the tip of the history.
Only modify the `useTxHistory` hook for filters.

Reply:
I'll revert it.

However, shouldn't we not poll if there is a filter query present? If the `txHistoryTag` changes, it will currently poll and clear the filtered results. I can see an edge case with this though: clearing pending transactions.

Reply:
It's fine if the filtered results will not self-update. We poll only to resolve pending states, and this will still work. Re-rendering the history is a nice side-effect, but it's not critical for a filtered view.